python_ta/debug/accumulation_table.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import annotations
+
+import inspect
+import sys
+
+import astroid
+import tabulate
+
+
+class AccumulationTable:
+    """
+    Class used as a form of print debugging to analyze different loop and
+    accumulation variables during each iteration in a for loop
+
+    Private instance attributes:
+        _num_iterations: the number of for loop iterations
+        _accumulation_names: the variable names of the different accumulation variables
+        _loop_accumulation_values: a mapping between the accumulation variables and their value during each iteration
+        _loop_var_name: the name of the loop variable
+        _loop_var_val: the value of the loop variable during each iteration
+        _curr_lineno: the current line number of the loop
+        _curr_values: the current accumulation values for this specific iteration
+
+    """
+
+    _num_iterations: int
+    _accumulation_names: list[str]
+    _loop_accumulation_values: dict[str, list]
+    _loop_var_name: str
+    _loop_var_val: list
+    _curr_lineno: int
+    _curr_values: list
+
+    def __init__(self, accumulation_names: list) -> None:
+        self._num_iterations = 0
+        self._accumulation_names = accumulation_names
+        self._loop_accumulation_values = {}
+        self._loop_var_name = ""
+        self._loop_var_val = []
```

Comment:
Don't use separate attributes for the accumulation variable values and the loop variable values. Keep one dictionary that stores all values.